### PR TITLE
Manifest package

### PR DESCRIPTION
**Pull request by cored**
The reason for this refactoring is to reduce the amount of integration test that we have right now. The test suite takes too long locally to run.

### CODE REVIEW

File: manifest/manifest.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+// ManifestExtension is added to each task manifest item to make a filename
+var ManifestExtension = ".yml"
```
Comment (therealplato) [inline manifest/manifest.go]: use my new ConfigType.ManifestExtension instead of hardcoding this please

Comment (cored) [inline manifest/manifest.go]: @therealplato Oh, I just copy and paste the files after your refactor; maybe you missed it? I'll do it, sure.

Comment (therealplato) [inline manifest/manifest.go]: yeah, whoops!

i really would prefer to have all business logic in services/manifest.go and only have the manifest file be a dumb type

Comment (cored) [inline manifest/manifest.go]: @therealplato I'm following another approach but similar to that idea. Take a look at the `instance` package. It have the definition of the data in there and also the operations that you can do on top of those structs. The idea is to be specific about the operations that work on top of a certain data. In my mind I think that at some point we are going to remove the `services` package I think it was a premature abstraction on my part to think in that package.

**Pull request closed without merging**